src/servers/http_server_v2.cc

```diff
@@ -1887,13 +1889,14 @@ HTTPAPIServerV2::EVBufferToInput(
       uint64_t offset = 0, byte_size = 0;
       const char* shm_region = nullptr;
       if (CheckSharedMemoryData(output, &shm_region, &offset, &byte_size)) {
-        if (output.FindMember("data") == output.MemberEnd()) {
-          return TRTSERVER_ErrorNew(
-              TRTSERVER_ERROR_INVALID_ARG,
-              "must not specify 'data' field in request output when using "
-              "shared "
-              "memory");
-        }
+        // Uncomment after client fix (Don't send data in case of shared memory)
+        // if (output.FindMember("data") == output.MemberEnd()) {
+        //   return TRTSERVER_ErrorNew(
+        //       TRTSERVER_ERROR_INVALID_ARG,
+        //       "must not specify 'data' field in request output when using "
+        //       "shared "
+        //       "memory");
+        // }
```

Comment:
Can be simply removed, requested output doesn't have "data" field anyway

Reply:
This is to verify that the infer request header does not contain 'data' fields. 
The docs specify that in the case of shared memory: the “data” field of request_input must not be set and the returned $response_output must not define the “data” field.

Reply:
You are checking the "output" in the request object which shouldn't have "data" field at all https://github.com/kubeflow/kfserving/blob/master/docs/predict-api/v2/required_api.md#request-output

Reply:
For output, the doc is referring to the response JSON message. You can add the check there if you want a sanity check, or just be sure that the "data" field will not be added in shared memory case.

Reply:
Fixed the check